netebpfext/net_ebpf_ext.c

```diff
@@ -615,6 +629,54 @@ net_ebpf_ext_uninitialize_ndis_handles()
     }
 }
 
+static void
+_net_ebpf_extension_cleanup_state(HANDLE wfp_engine_handle)
+/* ++
+
+   This function cleans up persisted WFP related state.
+
+-- */
+{
+    NET_EBPF_EXT_LOG_ENTRY();
+
+    ebpf_assert(wfp_engine_handle != NULL);
+
+    NTSTATUS status = STATUS_SUCCESS;
+    for (size_t index = 0; index < EBPF_COUNT_OF(_net_ebpf_ext_wfp_callout_states); index++) {
+        status = FwpmCalloutDeleteByKey(wfp_engine_handle, _net_ebpf_ext_wfp_callout_states[index].callout_guid);
+        if (!NT_SUCCESS(status)) {
+            NET_EBPF_EXT_LOG_MESSAGE_UINT64(
+                NET_EBPF_EXT_TRACELOG_LEVEL_ERROR,
+                NET_EBPF_EXT_TRACELOG_KEYWORD_EXTENSION,
+                "FwpmCalloutDeleteByKey failed for index#",
+                index);
+
+            NET_EBPF_EXT_LOG_NTSTATUS_API_FAILURE(
+                NET_EBPF_EXT_TRACELOG_KEYWORD_EXTENSION, "FwpmCalloutDeleteByKey", status);
+        }
+
+        status = FwpmSubLayerDeleteByKey(_wfp_engine_handle, _net_ebpf_ext_wfp_callout_states[index].layer_guid);
+        if (!NT_SUCCESS(status)) {
+            NET_EBPF_EXT_LOG_MESSAGE_UINT64(
```

Comment:
not needed.

Reply:
Not sure I understand.  Are you saying we don't need to do sublayer deletion? if yes, was curious as to why?